pkg/server/plugin/upstreamauthority/certmanager/certmanager_test.go

```diff
@@ -214,87 +202,178 @@ func Test_MintX509CA(t *testing.T) {
+		kube_config_file = "/path/to/config"
+		`,
+			expectConfig: &Config{
 				IssuerName:         "my-issuer",
 				IssuerKind:         "Issuer",
 				IssuerGroup:        "cert-manager.io",
 				Namespace:          "my-namespace",
 				KubeConfigFilePath: "/path/to/config",
 			},
+			expectConfigFile: "/path/to/config",
+		},
+		"no trust domain": {
+			inpConfig: `
+		issuer_name = "my-issuer"
+		namespace = "my-namespace"
+		kube_config_file = "/path/to/config"
+		`,
+			overrideCoreConfig: &catalog.CoreConfig{},
+			expectCode:         codes.InvalidArgument,
+			expectMsgPrefix:    "trust_domain is required",
+		},
+		"failed to create client": {
+			inpConfig: `
+		issuer_name = "my-issuer"
+		namespace = "my-namespace"
+		kube_config_file = "/path/to/config"
+		`,
+			newClientErr:     errors.New("some error"),
+			expectCode:       codes.Internal,
+			expectMsgPrefix:  "failed to create cert-manager client: some error",
+			expectConfigFile: "/path/to/config",
 		},
 	}
 
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
-			plugin := New()
-			plugin.SetLogger(hclog.Default())
+			var err error
+
+			options := []plugintest.Option{
+				plugintest.CaptureConfigureError(&err),
+				plugintest.Configure(test.inpConfig),
+			}
 
-			config, err := plugin.loadConfig(&spi.ConfigureRequest{
-				Configuration: test.inpConfig,
-			})
+			if test.overrideCoreConfig != nil {
+				options = append(options, plugintest.CoreConfig(*test.overrideCoreConfig))
+			} else {
+				options = append(options, plugintest.CoreConfig(catalog.CoreConfig{
+					TrustDomain: spiffeid.RequireTrustDomainFromString("localhost"),
+				}))
+			}
 
-			require.Equal(t, test.expErr, (err != nil))
-			if err != nil {
-				require.Equal(t, codes.InvalidArgument, status.Code(err))
+			p := &Plugin{
+				hooks: hooks{
+					newClient: func(configPath string) (client.Client, error) {
+						assert.Equal(t, test.expectConfigFile, configPath)
+						if test.newClientErr != nil {
+							return nil, test.newClientErr
+						}
+						return fakeclient.NewFakeClientWithScheme(scheme), nil
+					},
+				},
 			}
-			require.Equal(t, test.expConfig, config)
+
+			plugintest.Load(t, builtin(p), nil, options...)
+			spiretest.RequireGRPCStatusHasPrefix(t, err, test.expectCode, test.expectMsgPrefix)
+			if test.expectCode != codes.OK {
+				require.Nil(t, p.config)
+				require.Nil(t, p.cmclient)
+				return
+			}
+
+			require.Equal(t, test.expectConfig, p.config)
+			require.NotNil(t, p.cmclient)
 		})
 	}
 }
+
+func TestPublishJWTKey(t *testing.T) {
+	cmclient := fakeclient.NewFakeClientWithScheme(scheme)
+	crCreated := make(chan struct{}, 1)
+	staleCRsDeleted := make(chan struct{}, 1)
+
+	p := &Plugin{
+		hooks: hooks{
+			newClient: func(configPath string) (client.Client, error) {
+				return cmclient, nil
+			},
+			onCreateCR: func() {
+				crCreated <- struct{}{}
+			},
+			onCleanupStaleCRs: func() {
+				staleCRsDeleted <- struct{}{}
+			},
+		},
```

Comment:
do we need all these hooks to be set up for testing out the JWT key stuff?

Reply:
no, only cmclient. I'll remove anothers